format.js

```diff
@@ -0,0 +1,58 @@
+// consider this a warning about getting obsessive about optimization
+
+var utilformat = require('util').format
+
+
+function format (a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16) {
```

Comment:
Also if you cache `arguments.length` rather than putting it in the switch, that may also help. Hard to know without looking at IRHydra.